lib/redux/actions/async.js

```diff
@@ -0,0 +1,523 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2015, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+/* global chrome */
+
+import _ from 'lodash';
+import async from 'async';
+import semver from 'semver';
+
+import sundial from 'sundial';
+
+import * as actionTypes from '../constants/actionTypes';
+import * as actionSources from '../constants/actionSources';
+import { pages, paths, steps, urls } from '../constants/otherConstants';
+import { errorText } from '../utils/errors';
+
+import * as syncActions from './sync';
+import * as actionUtils from './utils';
+
+let services = {};
+let versionInfo = {};
+let daysForCareLink = null;
+
+/*
+ * ASYNCHRONOUS ACTION CREATORS
+ */
+
+export function doAppInit(config, servicesToInit) {
+  return (dispatch, getState) => {
+    dispatch(syncActions.setVersion(config.version));
+    services = servicesToInit;
+    versionInfo.semver = config.version;
+    versionInfo.name = config.namedVersion;
+    daysForCareLink = config.DEFAULT_CARELINK_DAYS;
+    const { api, carelink, device, localStore, log } = services;
+
+    dispatch(syncActions.initRequest());
+
+    async.series([
```

Comment:
Huh, the code I ported to redux was styled the way you suggest. I disprefer that style, but I guess I'm the minority since it was written your way before too. I'll change it.